### DISCUSSION TITLE
Need help to fix build issues of OPENSSL 3.0.7

### DISCUSSION THREAD

**Discussion by Shesh3799** [openssl/openssl#21584] [Q&A]
Hi All,
Actually I have a requirement in one of my project that I can't use the default lib names of openssl i.e libcrypto and libssl. So, I have modified the lib name to libxycrypto and libxyssl in all build.info file as well as in unix-Makefile.tmpl.

But after modifying the lib names I see that the makefile that is getting generated with ./Configure option is missing -I and -Iinclude option for various gcc commands in makefile.

Is there anything else that I have to modify to make things work?

Thanks,
Shesh

**Comment by paulidale** [comment]
OpenSSL is generally meant to be installed in system directories.  If this is done, there is no need for -I and -L options.

**Reply by Shesh3799** [reply]
Actually I want to ship openssl as part of binaries. So if I rename openssl it would help my binaries to differentiate between already existing openssl in system and the one I'll be shipping. The openssl version with which I have built my binaries might be different than the one present in system by default. Now to avoid this problem I'm renaming build.info files to generate a openssl with name libxyssl.so and libxycrypto.so
But after replacing the lib names the makeFile that is generated with ./Configure option is missing -I and -Iinclude  in gcc commands.

**Comment by nhorman** [comment]
can you post the changes you made, and the error that you are getting?

From your description it kind of sound like maybe the libraries are getting generated properly, but when you build your application you're getting failures, because you can't find the needed header files.  If thats the case my first guess would be that you forgot to modify the pkgconfig files that get installed with openssl that point application builds to where the headers and libraries are.  Take a look at the libcrypto.pc target in the openssl toplevel Makefile.  There you should see a line like:
echo 'Libs: -L$${libdir} -lcrypto'; \
you would need to change -lcrypto to -lxcrypto
repeat ad nasueum for the other libraries
